nav2_tasks/include/nav2_tasks/goal_reached_condition.hpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_TASKS__GOAL_REACHED_CONDITION_HPP_
+#define NAV2_TASKS__GOAL_REACHED_CONDITION_HPP_
+
+#include <string>
+#include <memory>
+#include "rclcpp/rclcpp.hpp"
+#include "behaviortree_cpp/condition_node.h"
+#include "nav2_robot/robot.hpp"
+#include "geometry_msgs/msg/pose_with_covariance_stamped.hpp"
+#include "nav2_tasks/compute_path_to_pose_task.hpp"
+
+namespace nav2_tasks
+{
+
+class GoalReachedCondition : public BT::ConditionNode
+{
+public:
+  explicit GoalReachedCondition(const std::string & condition_name)
+  : BT::ConditionNode(condition_name), initialized_(false)
+  {
+  }
+
+  GoalReachedCondition() = delete;
+
+  ~GoalReachedCondition()
+  {
+  }
+
+  BT::NodeStatus tick() override
+  {
+    if (!initialized_) {
+      node_ = blackboard()->template get<rclcpp::Node::SharedPtr>("node");
```

Comment:
I'd consider making these four lines a seperate function. Then the `tick` function will read very nicely

```
if (!initialized) {
  initialize();
}

if (goalReached()) ...
```
It's mostly just a problem with github interface, but it took me a few moments to realize that all four lines were part of initialize.

If you choose not to do this, I'd get rid of the blank lines between each statement.

Reply:
Fixed.